processor/signozspanmetricsprocessor/processor.go

```diff
@@ -15,88 +15,120 @@
 )
 
 const (
-	serviceNameKey             = conventions.AttributeServiceName
-	operationKey               = "operation" // is there a constant we can refer to?
-	spanKindKey                = "span.kind"
-	statusCodeKey              = "status.code"
-	TagHTTPStatusCode          = conventions.AttributeHTTPStatusCode
-	metricKeySeparator         = string(byte(0))
-	traceIDKey                 = "trace_id"
+	serviceNameKey     = conventions.AttributeServiceName
+	operationKey       = "operation"   // OpenTelemetry non-standard constant.
+	spanKindKey        = "span.kind"   // OpenTelemetry non-standard constant.
+	statusCodeKey      = "status.code" // OpenTelemetry non-standard constant.
+	tagHTTPStatusCode  = conventions.AttributeHTTPStatusCode
+	metricKeySeparator = string(byte(0))
+	traceIDKey         = "trace_id"
+
 	defaultDimensionsCacheSize = 1000
 	resourcePrefix             = "resource_"
 )
 
 var (
-	maxDuration   = time.Duration(math.MaxInt64)
-	maxDurationMs = durationToMillis(maxDuration)
-
 	defaultLatencyHistogramBucketsMs = []float64{
-		2, 4, 6, 8, 10, 50, 100, 200, 400, 800, 1000, 1400, 2000, 5000, 10_000, 15_000, maxDurationMs,
+		2, 4, 6, 8, 10, 50, 100, 200, 400, 800, 1000, 1400, 2000, 5000, 10_000, 15_000,
 	}
 )
 
+type exemplarData struct {
+	traceID pcommon.TraceID
+	spanID  pcommon.SpanID
+	value   float64
+}
+
 type metricKey string
 
 type processorImp struct {
-	lock   sync.RWMutex
+	lock   sync.Mutex
 	logger *zap.Logger
 	config Config
 
 	metricsExporter component.MetricsExporter
 	nextConsumer    consumer.Traces
 
 	// Additional dimensions to add to metrics.
-	callDimensions         []Dimension
-	dimensions             []Dimension
-	dbCallDimensions       []Dimension
-	externalCallDimensions []Dimension
+	dimensions             []dimension // signoz_latency metric
+	callDimensions         []dimension // signoz_calls_total metric
+	dbCallDimensions       []dimension // signoz_db_latency_* metric
+	externalCallDimensions []dimension // signoz_external_call_latency_* metric
+
 	// The starting time of the data points.
-	startTime time.Time
+	startTimestamp pcommon.Timestamp
 
-	// Call & Error counts.
-	callSum map[metricKey]int64
+	// Histogram.
+	histograms    map[metricKey]*histogramData // signoz_latency metric
+	latencyBounds []float64
 
-	// Latency histogram.
-	latencyCount        map[metricKey]uint64
-	latencySum          map[metricKey]float64
-	latencyBucketCounts map[metricKey][]uint64
+	callHistograms    map[metricKey]*histogramData // signoz_calls_total metric
+	callLatencyBounds []float64
 
-	latencyBounds []float64
+	dbCallHistograms    map[metricKey]*histogramData // signoz_db_latency_* metric
+	dbCallLatencyBounds []float64
 
-	dbLatencyCount map[metricKey]uint64
-	dbLatencySum   map[metricKey]float64
+	externalCallHistograms    map[metricKey]*histogramData // signoz_external_call_latency_* metric
+	externalCallLatencyBounds []float64
```

Comment:
we don't need percentiles of external calls, we are just showing `avg` duration in the charts. It was causing a lot of high cardinality data and hence I remove buckets from externalCalls earlier

Reply:
That is still the same behaviour; please take a look at `collectExternalCallMetrics` function. I left it here just in case we need it.

Reply:
ok